webui/react/src/components/kit/Select.tsx

```diff
@@ -0,0 +1,168 @@
+import { Select as AntdSelect } from 'antd';
+import type { DefaultOptionType, LabeledValue, RefSelectProps, SelectValue } from 'antd/es/select';
+import React, { forwardRef, useCallback, useMemo, useState } from 'react';
+
+import Label, { LabelTypes } from 'components/Label';
+import Icon from 'shared/components/Icon/Icon';
+
+import css from './Select.module.scss';
+
+const { OptGroup, Option } = AntdSelect;
+
+export { Option, SelectValue };
+
+type Options = DefaultOptionType | DefaultOptionType[];
+export interface Props<T = SelectValue> {
```

Comment:
question: do we need to use generics here? can we just add `SelectValue` without generics?
Also, `SelectProps` would be good to follow other UIkit convention

Reply:
A generic type is actually better since the type can vary.

Reply:
@ClaireNeveu 
i still think we can just use `SelectValue` directly instead of using generics.
The wrapped antd Select only accept `SelectValue` anyways, and the generic is used only for `value` and `defaultValue`. Even if we pass boolean or other unsupported type to generic, it wouldnt work.  Also, our current implementation doesn't  actually use generics because `Select` component doesn't accept generic. 
so i think we can remove generics.

```ts
type RawValue = string | number;

export interface LabeledValue {
    key?: string;
    value: RawValue;
    label: React.ReactNode;
}

export type SelectValue = RawValue | RawValue[] | LabeledValue | LabeledValue[] | undefined;
```

Reply:
We could actually use a constrained generic here and then cast appropriately and the interface would be correctly generic even if the underlying implementation isn't.

Reply:
i feel like the that makes select component a bit complicated, but also i think it depends on the constraints. 
Will we add the constraint into generics and make `Select` component generics in this PR?  
If so, what is the constraints for generics?

Reply:
It would only change the typing, not the actual logic. You could do `T extends SelectValue` and then use `T` instead of `SelectValue` letting you be more specific and avoid unnecessary type checks in handlers.

Reply:
make sense. i thought you were thinking about widening the acceptable types

Reply:
I generally prefer something like this be fully generic but AntD didn't have that foresight.